autorest/codegen/serializers/__init__.py

```diff
@@ -529,3 +546,162 @@ def _serialize_and_write_metadata(
         self._autorestapi.write_file(
             namespace_path / Path("_metadata.json"), metadata_serializer.serialize()
         )
+
+    def _package_root_folder(self, namespace_path: Path) -> Path:
```

Comment:
I feel like this is way too much code to add, not sure what this code is doing and it's hard to tell

Reply:
The function is to find where to put `generate_samples` which shall keep in same folder with `setup.py`. And we need to compatible with this kind of config: https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/purview/azure-purview-administration/swagger#readme